### ISSUE TITLE
osx-make-* attempts to build ObjC as CXX

### ISSUE THREAD

**Issue by code-disaster** [floooh/fips#34]
Building a small sample using fips-glfw. While osx-xcode-* works fine, osx-make-* fails with:

```
Building CXX object fips-glfw_glfw3/CMakeFiles/glfw3.dir/glfw/src/cocoa_monitor.m.o
[ 42%] error: invalid argument '-std=c++11' not allowed with 'C/ObjC'
```

Somehow CMake tries to compile those files as C++ / with C++ flags.

**Comment by floooh** [comment]
Strange, I saw a fix/hack for this in the original GLFW CMakeLists.txt file (https://github.com/glfw/glfw/blob/master/src/CMakeLists.txt#L55), but this wasn't necessary on my setup.

What Xcode/cmake version do you have? I'm at Xcode 6.1.1 and cmake version 3.1.0 (I suspect it's the cmake version where this might have been fixed in the meatime).

I'll add the fix to the fips-glfw makefile, I think that's better then requiring a bleeding edge cmake version.

**Comment by floooh** [comment]
It was easier to put the fix into fips, see here: 625878154126bc121bb48c3cf5f25256a63aaa3d, can you check if this works?

**Comment by code-disaster** [comment]
You were right about the cmake version. I've been using the latest macports version which is 3.0.2. I removed it and built cmake 3.1.0 from source. Now it compiles correctly.

The fix in fips.cmake didn't change anything. Still failed with 3.0.2, doesn't have any effect with 3.1.0.

**Comment by floooh** [comment]
Ok thanks for testing! I'll try to find a proper solution this evening with an older cmake version.

**Comment by floooh** [comment]
Doh, stupid bug in my 'fix' (setting the property on the file extension string, not the filename), anyway, I'll look into it in the evening :)

**Comment by floooh** [comment]
Should now really be fixed in 00430f7210a95daa836f2a1500c0f085a47e1aac, tested with cmake 2.8.11 on OSX. A './fips clean all' may be necessary.
